server/src/main/java/org/elasticsearch/search/aggregations/bucket/adjacency/InternalAdjacencyMatrix.java

```diff
@@ -228,6 +213,22 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
         return reduced;
     }
 
+    private InternalBucket reduceBucket(List<InternalBucket> buckets, ReduceContext context) {
```

Comment:
I'm commenting here, but may apply everywhere. Should the reduceBucket method be a part of the InternalMultiBucketAggregation as an abstract method? Just to help future development be contractually obligated to abide by this new pattern?